infrastructure/src/test/java/org/corfudb/infrastructure/CorfuSequencerServerTest.java

```diff
@@ -0,0 +1,528 @@
+     */
+    @Test
+    public void sequencerTrimRequest() {
+        // Bootstrap the Sequencer Server with a stream
+        UUID streamA = UUID.nameUUIDFromBytes("streamA".getBytes());
+        // Construct new tailMaps and create a to BootstrapSequencerRequestMsg
+        Map<UUID, StreamAddressSpace> tailMap = new HashMap<>();
+        long globalTail = 1;
+        // StreamAddressSpace streamAddressSpace = new StreamAddressSpace(Address.NON_ADDRESS,
+        //         Roaring64NavigableMap.bitmapOf(globalTail));
+        StreamAddressSpace streamAddressSpace = mock(StreamAddressSpace.class);
+        tailMap.put(streamA, streamAddressSpace);
+        long bootstrapMsgEpoch = 1;
+
+        SequencerServerCache cache = mock(SequencerServerCache.class);
+        SequencerServer.FactoryHelper mockFactoryHelper = mock(SequencerServer.FactoryHelper.class);
+        when(mockFactoryHelper.getStreamAddressSpaceMap()).thenReturn(tailMap);
+        // The global log tail is set to 0 when sequencer is bootstrapped with the above tailMap
+        when(mockFactoryHelper.getGlobalLogTail()).thenReturn(globalTail);
+        when(mockFactoryHelper.getSequencerServerCache(anyInt(), anyLong())).thenReturn(cache);
+        sequencerServer = new SequencerServer(mockServerContext, mockFactoryHelper);
+
+        // Send a request with SequencerTrimRequestMsg
+        long trimMark = 0;
+        RequestMsg request = getRequestMsg(
+                getBasicHeader(true, true),
+                CorfuProtocolSequencer.getSequencerTrimRequestMsg(trimMark)
+        );
+
+        // Sequencer epoch was set to bootstrapMsgEpoch in the previous RPC call.
+        when(mockServerContext.getServerEpoch()).thenReturn(bootstrapMsgEpoch);
+        sequencerServer.setSequencerEpoch(bootstrapMsgEpoch);
+
+        sequencerServer.handleMessage(request, mockChannelHandlerContext, mockServerRouter);
+
+        // Verify that sendResponse() was called and capture the response object
+        ArgumentCaptor<ResponseMsg> responseCaptor = ArgumentCaptor.forClass(ResponseMsg.class);
+        verify(mockServerRouter).sendResponse(responseCaptor.capture(),
+                any(ChannelHandlerContext.class));
+        ResponseMsg response = responseCaptor.getValue();
+
+        // Assert that the payload has a getSequencerMetricsResponseMsg and that the base
+        // header fields have remained the same
+        assertTrue(compareBaseHeaderFields(request.getHeader(), response.getHeader()));
+        // Assert that response has a sequencerTrimResponse object (Note: SequencerTrimResponse will not have any body)
+        assertTrue(response.getPayload().hasSequencerTrimResponse());
+        // Verify that sequencerServerCache was invalidated up to the trimMark
+        ArgumentCaptor<Long> trimMarkCaptor = ArgumentCaptor.forClass(Long.class);
+        verify(cache).invalidateUpTo(trimMarkCaptor.capture());
+        assertThat(trimMarkCaptor.getValue()).isEqualTo(trimMark);
```

Comment:
Is this argument captor necessary? I think calling verify with the expected object should suffice here. In most cases, if you know the result ahead of time, you don't need an argument captor.

Reply:
Done, Thank you.